stdlib/public/runtime/EnvironmentVariables.cpp

```diff
@@ -187,9 +191,32 @@ extern "C" char **_environ;
 #endif
 #endif
 
+#if defined(__ANDROID__)
+// On Android, also try loading runtime debug env variables from system props.
+static void platformInitialize(void *context) {
+  (void)context;
+#define SYSPROP_PREFIX "debug.swift.runtime."
+#define VARIABLE(name, type, defaultValue, help)                          \
+  do {                                                                    \
+    char name##_string[PROP_VALUE_MAX] = "";                              \
+    if (__system_property_get(SYSPROP_PREFIX #name, name##_string) > 0) { \
+      swift::runtime::environment::name##_isSet_variable = true;          \
+      swift::runtime::environment::name##_variable =                      \
+        parse_##type(#name, name##_string, defaultValue);                 \
+    }                                                                     \
+  } while (0);
+#include "EnvironmentVariables.def"
+#undef VARIABLE
+}
+#else
+static void platformInitialize(void *context) {
+  (void)context;
+}
+#endif
+
 #if !SWIFT_STDLIB_HAS_ENVIRON
 void swift::runtime::environment::initialize(void *context) {
-  (void)context;
+  platformInitialize(context);
 }
```

Comment:
I wonder if we should sink the android case under the `#if !SWIFT_STDLIB_HAS_ENVIRON` and avoid the `platformInitialize`.

Reply:
`SWIFT_STDLIB_HAS_ENVIRON` and `ENVIRON` are both defined when building for Android. I added the `platformInitialize` call to all three locations for completeness.

Not sure if it impacts your thinking, but it is useful to support both env vars and sysprops on Android because command line apps still work fine with env vars.

Reply:
Okay, that makes sense. This just adds a platform specific hook, and then uses that for android.